src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -262,6 +262,7 @@ private int readChunkCount(DataInput input)
     private Memory readChunkOffsets(DataInput input, int startIndex, int endIndex)
     {
         final int allChunkCount = readChunkCount(input);
+        endIndex = Math.min(allChunkCount, endIndex + 1); // add one to make sure we get one more chunk in order to be able to calculate next chunk offset and thus the size
```

Comment:
Don't we have compressed file length for this?

Reply:
yes, but I'll add an additional check for that and throw corrupted sstable exception if it does not hold